SU2_CFD/src/solvers/CIncEulerSolver.cpp

```diff
@@ -2899,6 +2907,19 @@ void CIncEulerSolver::SetResidual_DualTime(CGeometry *geometry, CSolver **solver
     AD::EndNoSharedReading();
   }
 
+
+  // update the density and cp
+  SU2_OMP_FOR_STAT(omp_chunk_size)
+  for (iPoint = 0; iPoint < nPointDomain; iPoint++) {      // set density of time n to density of time n+1
+    Density = nodes->GetDensity(iPoint);
+    Cp = nodes->GetSpecificHeatCp(iPoint);
+
+    nodes->SetDensity_time_n(iPoint, Density);
+    nodes->SetCp_time_n(iPoint, Cp);
+  }
+  END_SU2_OMP_FOR
+
+
```

Comment:
What would be the best place to update this?

Reply:
Hi Cristopher, that is possible to reconstruct rho, but we also need rho if we do not have species transport. We could then construct it from the ideal gas law. But instead of recomputing with different methods, we could store the field. I think it is more generally applicable and less prone to error.